manager.go

```diff
@@ -356,8 +356,10 @@ func (m *DefaultExtensionManager) RegisterExtensions() error {
 		webhooks = append(webhooks, w)
 	}
 
-	if err := m.WebhookConfig.registerWebhooks(m.Context, webhooks); err != nil {
-		return errors.Wrap(err, "generating the webhook server configuration")
+	if m.Options.RegisterWebHook == nil || m.Options.RegisterWebHook != nil && *m.Options.RegisterWebHook {
```

Comment:
After a discussion with @mudler , making this code more readable needs a heavy refactoring. For now we will add a comment on what these conditions can be used for. We will plan a refactoring to make this whole method more readable and with clear responsibility.